streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -146,15 +148,25 @@ public StreamTask(final TaskId id,
         // initialize the consumed and committed offset cache
         consumedOffsets = new HashMap<>();
 
-        // create queues for each assigned partition and associate them
-        // to corresponding source nodes in the processor topology
-        final Map<TopicPartition, RecordQueue> partitionQueues = new HashMap<>();
+        defaultTimestampExtractor = config.defaultTimestampExtractor();
+        defaultDeserializationExceptionHandler = config.defaultDeserializationExceptionHandler();
 
         // initialize the topology with its own context
         processorContext = new ProcessorContextImpl(id, this, config, this.recordCollector, stateMgr, streamsMetrics, cache);
 
-        final TimestampExtractor defaultTimestampExtractor = config.defaultTimestampExtractor();
-        final DeserializationExceptionHandler defaultDeserializationExceptionHandler = config.defaultDeserializationExceptionHandler();
+        recordInfo = new PartitionGroup.RecordInfo();
+        partitionGroup = new PartitionGroup(createPartitionQueues(logContext),
+                                            TaskMetrics.recordLatenessSensor(threadId, taskId, streamsMetrics));
+
+        stateMgr.registerGlobalStateStores(topology.globalStateStores());
+    }
+
+    // create queues for each assigned partition and associate them
+    // to corresponding source nodes in the processor topology
+    private Map<TopicPartition, RecordQueue> createPartitionQueues(final LogContext logContext) {
+        final Set<TopicPartition> partitions = inputPartitions();
```

Comment:
We could just call `inputPartitions` on L170

Reply:
Yes, inlined this